config/coreboot-galp5.config

```diff
@@ -0,0 +1,880 @@
+CONFIG_HAVE_INTEL_FSP_REPO=y
+CONFIG_ADD_FSP_BINARIES=y
+CONFIG_FSP_T_LOCATION=0xfffe0000
+CONFIG_FSP_S_CBFS="fsps.bin"
+CONFIG_FSP_M_CBFS="fspm.bin"
+CONFIG_FSP_FULL_FD=y
+CONFIG_FSP_T_RESERVED_SIZE=0x0
+CONFIG_FSP_M_XIP=y
+CONFIG_FSP_USES_CB_STACK=y
+CONFIG_FSP_COMPRESS_FSP_S_LZ4=y
+CONFIG_FSP_STATUS_GLOBAL_RESET_REQUIRED_3=y
+CONFIG_FSP_STATUS_GLOBAL_RESET=0x40000003
+CONFIG_SOC_INTEL_COMMON_FSP_RESET=y
+CONFIG_FSPS_USE_MULTI_PHASE_INIT=y
+CONFIG_USE_FSP_NOTIFY_PHASE_POST_PCI_ENUM=y
+CONFIG_USE_FSP_NOTIFY_PHASE_READY_TO_BOOT=y
+CONFIG_USE_FSP_NOTIFY_PHASE_END_OF_FIRMWARE=y
+# CONFIG_DISPLAY_FSP_TIMESTAMPS is not set
+CONFIG_INTEL_GMA_ACPI=y
+CONFIG_VBT_CBFS_COMPRESSION_LZMA=y
+# CONFIG_VBT_CBFS_COMPRESSION_LZ4 is not set
+# CONFIG_VBT_CBFS_COMPRESSION_NONE is not set
+CONFIG_VBT_CBFS_COMPRESSION_ALGORITHM="lzma"
+CONFIG_INTEL_GMA_VERSION_2=y
+CONFIG_DRIVERS_INTEL_PMC=y
+# CONFIG_DRIVERS_NXP_UWB_SR1XX is not set
+# CONFIG_DRIVERS_PS2_KEYBOARD is not set
+CONFIG_DRIVERS_MC146818=y
+CONFIG_USE_PC_CMOS_ALTCENTURY=y
+CONFIG_PC_CMOS_BASE_PORT_BANK0=0x70
+CONFIG_MEMORY_MAPPED_TPM=y
+CONFIG_TPM_TIS_BASE_ADDRESS=0xfed40000
+# CONFIG_DRIVERS_SIL_3114 is not set
+CONFIG_DRIVERS_USB_ACPI=y
+CONFIG_DRIVERS_WIFI_GENERIC=y
+# CONFIG_DRIVERS_MTK_WIFI is not set
+CONFIG_MP_SERVICES_PPI=y
+CONFIG_MP_SERVICES_PPI_V1=y
+CONFIG_DRIVERS_INTEL_USB4_RETIMER=y
+# end of Generic Drivers
+
+#
+# Security
+#
+
+#
+# CBFS verification
+#
+# CONFIG_CBFS_VERIFICATION is not set
+# end of CBFS verification
+
+#
+# Verified Boot (vboot)
+#
+CONFIG_VBOOT_LIB=y
+# end of Verified Boot (vboot)
+
+#
+# Trusted Platform Module
+#
+# CONFIG_NO_TPM is not set
+CONFIG_TPM2=y
+CONFIG_TPM=y
+CONFIG_MAINBOARD_HAS_TPM2=y
+# CONFIG_DEBUG_TPM is not set
+CONFIG_TPM_RDRESP_NEED_DELAY=y
+# CONFIG_TPM_LOG_CB is not set
+CONFIG_TPM_LOG_TPM2=y
+# CONFIG_TPM_HASH_SHA1 is not set
+CONFIG_TPM_HASH_SHA256=y
+# CONFIG_TPM_HASH_SHA384 is not set
+# CONFIG_TPM_HASH_SHA512 is not set
+CONFIG_TPM_MEASURED_BOOT_RUNTIME_DATA=""
+CONFIG_PCR_BOOT_MODE=1
+CONFIG_PCR_HWID=1
+CONFIG_PCR_SRTM=2
+CONFIG_PCR_FW_VER=10
+CONFIG_PCR_RUNTIME_DATA=3
+# end of Trusted Platform Module
+
+#
+# Memory initialization
+#
+CONFIG_PLATFORM_HAS_DRAM_CLEAR=y
+# CONFIG_SECURITY_CLEAR_DRAM_ON_REGULAR_BOOT is not set
+# end of Memory initialization
+
+# CONFIG_INTEL_TXT is not set
+# CONFIG_STM is not set
+# CONFIG_INTEL_CBNT_SUPPORT is not set
+CONFIG_BOOTMEDIA_LOCK_NONE=y
+# CONFIG_BOOTMEDIA_LOCK_CONTROLLER is not set
+# CONFIG_BOOTMEDIA_LOCK_CHIP is not set
+# CONFIG_BOOTMEDIA_SMM_BWP is not set
+# end of Security
+
+CONFIG_ACPI_HAVE_PCAT_8259=y
+CONFIG_ACPI_INTEL_HARDWARE_SLEEP_VALUES=y
+CONFIG_ACPI_SOC_NVS=y
+CONFIG_ACPI_CUSTOM_MADT=y
+CONFIG_ACPI_NO_CUSTOM_MADT=y
+CONFIG_ACPI_COMMON_MADT_LAPIC=y
+CONFIG_ACPI_COMMON_MADT_IOAPIC=y
+CONFIG_HAVE_ACPI_TABLES=y
+CONFIG_ACPI_LPIT=y
+CONFIG_BOOT_DEVICE_SPI_FLASH=y
+CONFIG_BOOT_DEVICE_MEMORY_MAPPED=y
+CONFIG_BOOT_DEVICE_SUPPORTS_WRITES=y
+CONFIG_RTC=y
+CONFIG_HEAP_SIZE=0x100000
```

Comment:
Might need to be bumped to 0x100000 otherwise no bootsplash

Reply:
@loftlifter31 added under [be1f40b](https://github.com/linuxboot/heads/pull/1774/commits/be1f40b8343de0a3bf9daf0fa04c06c26181b380)

Others might be needed see commit message

Reply:
Are we trying to get galp5 for your sole usage or for everybody else? That will change my involvement in this effort.

Reply:
My primary concern at this point is to get it working for myself according to my needs. However, I don't see any problem with testing alternate configurations which may benefit others, such as increasing the heap size.

Reply:
Put at right value under [605386](https://github.com/linuxboot/heads/pull/1774/commits/860538611fcca62e4e522883fdfd1c037c46434a) with patch under patches/ see pr files changed.

Reply:
I rebuilt heads with CONFIG_HEAP_SIZE=0x100000. Still no bootsplash at this size. Above you suggested a patch may be needed to increase to 400000 however the patch you provided is for coreboot version 24.02.01. I'm using version 4.22.01 for my build.

Reply:
Why would you use 24.02.01 and not system76 latest know commit?

https://github.com/coreboot/coreboot/compare/0a280ff747b44d7a2a62206ea0c25dbcbc5ce20b...6da534f92044c684139675a599eef92d077b2b36

- 24.02.01: https://github.com/coreboot/coreboot/commit/0a280ff747b44d7a2a62206ea0c25dbcbc5ce20b
- system76 fork pointed by this RP modules/coreboot fork : https://github.com/system76/coreboot/commit/6da534f92044c684139675a599eef92d077b2b36

As said, yes, the patch that was missing is under patches to be applied under clean build under this PR: https://github.com/linuxboot/heads/blob/1e840fcb4b397ea3e87f40abcb961a41965de155/patches/coreboot-system76-unreleased/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch

You would need to rebuild this PR with modules/coreboot helper:
```
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5 real.remove_canary_files-extract_patch_rebuild_what_changed
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5
```

Otherwise the value cannot be changed in config file.

@loftlifter31 Please open a PR for what you attempt to do. This PR uses system76 coreboot branch so that whatever bug you find along the way will be possible to report to them, leaving me only playing with Heads parts, delegating responsibilities to system76 if things coreboot don't work: just like this missing patch and others missing under patches/coreboot-24.02.01 for them to pick up. You are more then invited into pinging them to do so, linking them to the PR that made those patches landing (use github blame/history for a file in the file browser https://github.com/linuxboot/heads/tree/master/patches/coreboot-24.02.01 then clicking https://github.com/linuxboot/heads/blob/master/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch, look left corner, expend commit message -> leads to https://github.com/linuxboot/heads/commit/119cebc44f51b0d69069bf37e9817156b96df09c, clikc PR number associated under commit message - > https://github.com/linuxboot/heads/pull/1723


System76 won't offer support for an older version. If you want them to care, you have to use and refer to their head commit, which is https://github.com/system76/coreboot/commit/6da534f92044c684139675a599eef92d077b2b36 for which those 2 patches are missing (they have to cherry-pick them and adapt them):
- https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0001-prevent_bootsplash_heap_allocation_failure_to_booting.patch
- https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch

---

TLDR:

Do 
```
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5 real.remove_canary_files-extract_patch_rebuild_what_changed
docker run -e DISPLAY=$DISPLAY --network host --rm -ti -v $(pwd):$(pwd) -w $(pwd) tlaurion/heads-dev-env:latest -- make -d BOARD=galp5
```

from this branch. This will apply https://github.com/linuxboot/heads/blob/119cebc44f51b0d69069bf37e9817156b96df09c/patches/coreboot-24.02.01/0002-increase_heap_from1mb_to4mb_when_bootpslash_enabled.patch which will permit you to increase heap size and you will get boosplash.

Note that system76 still need to apply those patches: please open an issue on their side.
Also note that if someone was raising an issue for an older version of Heads: I would say to them to use latest commit and would stop caring there. Expect System76 to do the same. Talk to them about their latest commit. This is why I created this PR to aid in your journay: to anyone galp5 wanting to use Heads to be able to do so once this PR is merged.

You continuing your adventure on your local fork without pushing a branch to have eyes on.... is to say the least, non-interesting and lowers my interests helping you, helping me, helping galp5 to get heads. Thanks for your understanding.

Reply:
Ha, nope. Sorry, coreboot forks have their own logic, and doing real.clean* doesn't remove the .patched flag.
Working on a fix and pushing.

Reply:
Ha. Realized the two patches were abandoned and https://review.coreboot.org/c/coreboot/+/83895 was merged instead.... @JonathonHall-Purism

Reply:
See [127e5df](https://github.com/linuxboot/heads/pull/1774/commits/127e5df6dadcc9ecbcd3c8f29e9e755f131c38b7) and https://github.com/linuxboot/heads/pull/1795

There should be no need to increase HEAP anymore.

Reply:
Neglect https://github.com/linuxboot/heads/pull/1774#discussion_r1755582417 referred patches, deleted in latest ommit here and under https://github.com/linuxboot/heads/pull/1795

Reply:
Still no bootsplash with heap set to 0x400000